pty/ptytest/ptytest_test.go

```diff
@@ -15,4 +21,41 @@ func TestPtytest(t *testing.T) {
 		pty.ExpectMatch("write")
 		pty.WriteLine("read")
 	})
+
+	t.Run("Cobra ptytest should not hang when output is not consumed", func(t *testing.T) {
+		t.Parallel()
+
+		tests := []struct {
+			name          string
+			output        string
+			isPlatformBug bool // See https://github.com/coder/coder/issues/2122 for more info.
+		}{
+			{name: "1024 is safe (does not exceed macOS buffer)", output: strings.Repeat(".", 1024)},
+			{name: "1025 exceeds macOS buffer (must not hang)", output: strings.Repeat(".", 1025), isPlatformBug: true},
+			{name: "10241 large output", output: strings.Repeat(".", 10241), isPlatformBug: true}, // 1024 * 10 + 1
+		}
+		for _, tt := range tests {
+			tt := tt
+			// nolint:paralleltest // Avoid parallel test to more easily identify the issue.
+			t.Run(tt.name, func(t *testing.T) {
+				if tt.isPlatformBug && (runtime.GOOS == "darwin" || runtime.GOOS == "windows") {
+					t.Skip("This test hangs on macOS and Windows, see https://github.com/coder/coder/issues/2122")
+				}
```

Comment:
If this fails on mac, I'm good with merging it.

Reply:
Maybe push a commit without the skip to confirm it fails on ci?

Reply:
> If this fails on mac, I'm good with merging it.

Yup, it fails on my Intel Mac (and CI), don't know about others 😄.

> Maybe push a commit without the skip to confirm it fails on ci?

I did in earlier commits: https://github.com/coder/coder/runs/6771839024?check_suite_focus=true